File: plant-back/PlantLibrary/Models/Question.cs
using PlantLibrary.Enums;
using SharedLibrary.Database;
using System;
using System.Collections.Generic;

namespace PlantLibrary.Models
{
    [BsonCollection("Question")]
    public class Question : Document
    {
        public QuestionType Type { get; set; }
        public string Title { get; set; }
        public string Text { get; set; }
        public bool IsClosed { get; set; } // Default : false
        public List<Reply> Replies { get; set; }

        // PICTURES
        public List<Picture> Pictures { get; set; }

        //RELATIONS
        public User User { get; set; }
        public Plant Plant { get; set; }

    }


}

File: plant-front/src/models/question.ts
import {QuestionType} from '../enums/question-type.enum';

export interface IQuestion {
  id: number;
  date: Date;
  userID: number;
  type: QuestionType;
  plantID: number;
  title: string;
  text: string;
  isClosed: boolean;
}

export class Question {
  Date: Date;
  Id: number;
  IsClosed: boolean;
  PlantID: number;
  Text: string;
  Title: string;
  Type: QuestionType;
  UserID: number;

  constructor(input: IQuestion) {
    this.Date = input.date;
    this.Id = input.id;
    this.IsClosed = input.isClosed;
    this.PlantID = input.plantID;
    this.Text = input.text;
    this.Title = input.title;
    this.Type = input.type;
    this.UserID = input.userID;
  }
}
